web/thesauruses/javascript/ECMAScript 2009/functions.json

```diff
@@ -0,0 +1,113 @@
+      "code": [
+        "function method(param1) {}",
+        "function method(param1, param2) {}"
+      ]
+    },
+    "void_function_with_keyword_parameters": {
+      "name": "Function that does not return a value and that takes 0 or more defined keyword parameters",
+      "code": [
+        "function method({param1}) {}",
+        "function method({param1, param2}) {}"
+      ]
+    },
+    "void_function_variable_parameters": {
+      "name": "Function that does not return a value and function that takes an unknown number of parameters",
+      "code": [
+        "function method() {",
+        "  var params = Array.prototype.slice.call(arguments); // convert arguments to an array",
+        "  // do something with params",
+        "}"
+      ]
+    },
+    "return_value_function_no_parameters": {
+      "name": "Function that returns a value and takes no parameters",
+      "code": "function method() { return true; }"
+    },
+    "return_value_function_with_parameters": {
+      "name": "Function that returns a value and takes 1 or more defined parameters",
+      "code": [
+        "function method(param1) { return true; }",
+        "function method(param1, param2) { return true; }"
+      ]
+    },
+    "return_value_function_with_keyword_parameters": {
+      "name": "Function that returns a value and takes 0 or more defined keyword parameters",
+      "code": "function method({param1, param2}) { return true; }"
+    },
+    "return_value_function_variable_parameters": {
+      "name": "Function that returns a value and takes an unknown number of parameters",
+      "code": [
+        "function method() {",
+        "  var params = Array.prototype.slice.call(arguments); // convert arguments to an array",
+        "  // do something with params",
+        "  return true;",
+        "}"
+      ]
+    },
+    "anonymous_function_no_parameters": {
+      "name": "Anonymous function that takes no parameters",
+      "code": "var method = function() {}"
+    },
+    "anonymous_function_with_parameters": {
+      "name": "Anonymous function that takes 1 or more defined parameters",
+      "code": "var method = function(param1, param2) {}"
+    },
+    "anonymous_function_with_keyword_parameters": {
+      "name": "Anonymous function that takes 0 or more defined keyword parameters",
+      "code": "var method = function({param1, param2}) {}"
+    },
+    "anonymous_function_variable_parameters": {
+      "name": "Anonymous function that takes an unknown number of parameters",
+      "code": [
+        "var method = function() {",
+        "  var params = Array.prototype.slice.call(arguments); // convert arguments to an array",
+        "  // do something with params",
+        "}"
+      ]
+    },
+    "anonymous_function_no_parameters_with_return": {
+      "name": "Anonymous function that takes no parameters and returns a value",
+      "code": "var method = function() { return true; }"
```

Comment:
I know there's other forms like `() => { statement; }`. Could you include those as well?